test/common/buffer/buffer_speed_test.cc

```diff
@@ -113,8 +113,9 @@ static void BufferPrependBuffer(benchmark::State& state) {
     // buffer every time without the overhead of a copy, we use an BufferFragment that references
     // (and never deletes) an external string.
     Buffer::OwnedImpl to_add;
-    Buffer::BufferFragmentImpl fragment(input.data(), input.size(), DoNotReleaseFragment);
-    to_add.addBufferFragment(fragment);
+    auto fragment =
+        std::make_unique<Buffer::BufferFragmentImpl>(input.data(), input.size(), DeleteFragment);
+    to_add.addBufferFragment(*fragment.release());
```

Comment:
Can you separate out the general "make benchmarks more awesome/readable/correct" from the Bazel aspects, ideally this PR is just Bazel.

Reply:
I can certainly make the benchmark tests blindly first, could I get a fast review on that?

I think the main blocker on this CL right now is the compile_time_options that I can't fully replicate.  There may be some existing breakages in that mode.

Reply:
Yes, happy to provide fast turnaround on PRs here. For compile_time_options, I wonder if `--build_test_only` is interacting somehow in https://github.com/envoyproxy/envoy/blob/master/ci/do_ci.sh#L233?